erc1155/src/lib.rs

```diff
@@ -0,0 +1,476 @@
+// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.4.2/contracts/token/ERC1155/IERC1155.sol
+
+#![no_std]
+#![feature(const_btree_new)]
+
+#[cfg(test)]
+use codec::{Decode, Encode};
+use gstd::{debug, exec, msg, prelude::*, ActorId};
+use primitive_types::U256;
+use scale_info::TypeInfo;
+
+pub mod base;
+
+const GAS_RESERVE: u64 = 500_000_000;
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+
+#[derive(Debug)]
+struct Erc1155Token {
+    name: String,
+    symbol: String,
+    description: String,
+    base_uri: String,
+    balances: BTreeMap<u128, BTreeMap<ActorId, u128>>,
+    operator_approvals: BTreeMap<ActorId, BTreeMap<ActorId, bool>>,
+}
+
+static mut ERC1155_TOKEN: Erc1155Token = Erc1155Token {
+    name: String::new(),
+    symbol: String::new(),
+    base_uri: String::new(),
+    description: String::new(),
+    balances: BTreeMap::new(),
+    operator_approvals: BTreeMap::new(),
+};
+
+impl Erc1155Token {
+    fn get_balance(&self, account: &ActorId, id: &u128) -> u128 {
+        *self
+            .balances
+            .get(id)
+            .and_then(|m| m.get(account))
+            .unwrap_or(&0)
+    }
+
+    fn set_balance(&mut self, account: &ActorId, id: &u128, amount: u128) {
+        debug!(
+            "before mint: {:?}, id: {:?}",
+            self.get_balance(account, id),
+            id
+        );
+
+        let mut _balance = self
+            .balances
+            .entry(*id)
+            .or_default()
+            .insert(*account, amount);
+
+        debug!(
+            "after mint: {:?}, id: {:?}",
+            self.get_balance(account, id),
+            id
+        );
+    }
+
+    fn balance_of(&self, account: &ActorId, id: &u128) -> u128 {
+        self.get_balance(account, id)
+    }
+
+    fn balance_of_batch(&self, accounts: &[ActorId], ids: &[u128]) -> Vec<BalanceOfBatchReply> {
+        if accounts.len() != ids.len() {
+            panic!("ERC1155: accounts and ids length mismatch");
+        }
+
+        let mut arr: Vec<BalanceOfBatchReply> = Vec::new();
+
+        for (i, ele) in ids.iter().enumerate() {
+            let account = accounts[i];
+            let amount = self.get_balance(&account, &ele);
+
+            let obj = BalanceOfBatchReply {
+                account: account,
+                id: *ele,
+                amount: amount,
+            };
+
+            arr.push(obj);
+        }
+
+        return arr;
+    }
+
+    fn mint(&mut self, from: &ActorId, id: &u128, amount: u128) {
+        // check owner
+        if from == &ZERO_ID {
```

Comment:
I think it it better to name the variable `to` instead of `from
```suggestion
        if to == &ZERO_ID {
```